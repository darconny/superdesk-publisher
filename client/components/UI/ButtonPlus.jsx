import React from 'react'
import PropTypes from 'prop-types'

const ButtonPlus = ({ onClick }) => {
    return (
        <button className="btn btn--primary btn--icon-only-circle" onClick={onClick}>
            <i className="icon-plus-large"></i>
        </button>
    )
}

ButtonPlus.propTypes = {
    onClick: PropTypes.func
}

export default ButtonPlus