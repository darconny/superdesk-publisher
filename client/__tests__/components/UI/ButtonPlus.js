import React from "react";
import ButtonPlus from "../../../components/UI/ButtonPlus";
import { render, fireEvent } from "@testing-library/react";

describe("UI/ButtonPlus", () => {
  it("renders correctly", () => {
    const { container } = render(<ButtonPlus />);

    expect(container.firstChild).toMatchSnapshot();
  });

  it("onClick function fired", () => {
    const onClick = jest.fn();
    const { container } = render(<ButtonPlus onClick={onClick} />);
    const button = container.querySelector("button");

    fireEvent.click(button);
    expect(onClick).toHaveBeenCalled();
  });
});
