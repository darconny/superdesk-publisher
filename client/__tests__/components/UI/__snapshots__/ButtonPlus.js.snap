// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UI/ButtonPlus renders correctly 1`] = `
<button
  class="btn btn--primary btn--icon-only-circle"
>
  <i
    class="icon-plus-large"
  />
</button>
`;
